compiler/rustc_typeck/src/check/method/probe.rs

```diff
@@ -765,12 +765,11 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
             )
         });
 
-        // It is illegal to invoke a method on a trait instance that
-        // refers to the `Self` type. An error will be reported by
-        // `enforce_object_limitations()` if the method refers to the
-        // `Self` type anywhere other than the receiver. Here, we use
-        // a substitution that replaces `Self` with the object type
-        // itself. Hence, a `&self` method will wind up with an
+        // It is illegal to invoke a method on a trait instance that refers to
+        // the `Self` type. An error will be reported by `object_safety.rs` if
```

Comment:
Wow, I had to do some archeology for this! The comment was first introduced in 1bceb98084b0e1010d123f48785a89bc7ad7e40f all the way back in 2013. `fn enforce_object_limitations` was removed in 8d8d8d4e5292c2fa4a622d981a5f85fd3d8f34d0 and replaced by `fn check_object_safety`, but the comment was never updated. Then `check_object_safety` gets removed in 843db01bd925279da0a56efde532c9e3ecf73610 and the check is moved to `report_selection_error` in src/librustc/middle/traits/error_reporting.rs. That got me far enough that I found that the error is `SupertraitSelf` and now lives in `compiler/rustc_trait_selection/src/traits/object_safety.rs`.

In short I think it would be helpful to mention `SupertraitSelf` so other people can find this in the future.
```suggestion
        // the `Self` type. [`ObjectSafetyViolation::SupertraitSelf`] will be reported by `object_safety.rs` if
```